redisinsight/api/src/modules/browser/services/string-business/string-business.service.ts

```diff
@@ -68,10 +71,64 @@ export class StringBusinessService {
 
   public async getStringValue(
     clientMetadata: ClientMetadata,
-    dto: GetKeyInfoDto,
+    dto: GetStringInfoDto,
   ): Promise<GetStringValueResponse> {
     this.logger.log('Getting string value.');
 
+    const { keyName, start, end } = dto;
+    let result: GetStringValueResponse;
+
+    try {
+      let value;
+      if (end) {
+        value = await this.browserTool.execCommand(
+          clientMetadata,
+          BrowserToolStringCommands.Getrange,
+          [keyName, `${start}`, `${end}`],
+        );
+      } else {
+        value = await this.browserTool.execCommand(
+          clientMetadata,
+          BrowserToolStringCommands.Get,
+          [keyName],
+        );
+      }
+
+      result = { value, keyName };
+    } catch (error) {
+      this.logger.error('Failed to get string value.', error);
+      if (error.message.includes(RedisErrorCodes.WrongType)) {
+        throw new BadRequestException(error.message);
+      }
+      catchAclError(error);
+    }
+
+    if (result.value === null) {
+      this.logger.error(
+        `Failed to get string value. Not Found key: ${keyName}.`,
+      );
+      throw new NotFoundException();
+    }
+
+    this.recommendationService.check(
+      clientMetadata,
+      RECOMMENDATION_NAMES.STRING_TO_JSON,
+      { value: result.value, keyName: result.keyName },
+    );
+    this.logger.log('Succeed to get string value.');
+
+    return plainToClass(GetStringValueResponse, result);
+  }
+
+  public async downloadStringValue(
+    clientMetadata: ClientMetadata,
+    dto: GetKeyInfoDto,
+  ): Promise<{ stream: Readable }> {
+    // const result = await this.getStringValue(
+    //   clientMetadata,
+    //   dto,
+    // );
+
```

Comment:
can we use commented function here instead of duplicating getStringValue method implementation?